tests/test_config_command.rs

```diff
@@ -0,0 +1,130 @@
+// limitations under the License.
+
+use itertools::Itertools;
+use regex::Regex;
+
+use crate::common::TestEnvironment;
+
+pub mod common;
+
+#[test]
+fn test_config_list_single() {
+    let mut test_env = TestEnvironment::default();
+    test_env.remove_env_var("JJ_USER");
+    test_env.add_config(
+        r###"
+    [user]
+    name = "Some User"
+    "###
+        .as_bytes(),
+    );
+
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "list", "user.name"]);
+    insta::assert_snapshot!(stdout, @r###"
+    user.name="Some User"
+    "###);
+}
+
+#[test]
+fn test_config_list_table() {
+    let mut test_env = TestEnvironment::default();
+    test_env.remove_env_var("JJ_USER");
+    test_env.remove_env_var("JJ_EMAIL");
+    test_env.add_config(
+        r###"
+    [user]
+    name = "Some User"
+    email = "some.user@example.com"
+    "###
+        .as_bytes(),
+    );
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "list", "user"]);
+    // TODO: Remove sorting when config crate maintains deterministic ordering.
+    insta::assert_snapshot!(
+        stdout.lines().sorted().join("\n"),
+        @r###"
+    user.email="some.user@example.com"
+    user.name="Some User"
+    user.timestamp="2001-02-03T04:05:07+07:00"
+    "###);
+}
+
+#[test]
+fn test_config_list_array() {
+    let test_env = TestEnvironment::default();
+    test_env.add_config(
+        r###"
+    test-array = [1, "b", 3.4]
+    "###
+        .as_bytes(),
+    );
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "list", "test-array"]);
+    insta::assert_snapshot!(stdout, @r###"
+    test-array=[1, "b", 3.4]
+    "###);
+}
+
+#[test]
+fn test_config_list_inline_table() {
+    let test_env = TestEnvironment::default();
+    test_env.add_config(
+        r###"
+        [[test-table]]
+        x = 1
+        [[test-table]]
+        y = ["z"]
+    "###
+        .as_bytes(),
+    );
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "list", "test-table"]);
+    insta::assert_snapshot!(stdout, @r###"
+    test-table=[{x=1}, {y=["z"]}]
+    "###);
+}
+
+#[test]
+fn test_config_list_all() {
+    let mut test_env = TestEnvironment::default();
+    test_env.remove_env_var("JJ_USER");
+    test_env.remove_env_var("JJ_EMAIL");
+    test_env.remove_env_var("JJ_OP_USERNAME");
+    test_env.remove_env_var("JJ_OP_HOSTNAME");
+    test_env.add_config(
+        r###"
+    [user]
+    name = "Some User"
+    [operation]
+    username = "some-username"
+    hostname = "some.host.example.com"
+    "###
+        .as_bytes(),
+    );
+    let stdout = test_env.jj_cmd_success(test_env.env_root(), &["config", "list"]);
+    // TODO: Remove sorting when config crate maintains deterministic ordering.
+    insta::assert_snapshot!(
+        find_stdout_lines(r"(operation|user)\.[^=]+", &stdout).iter().sorted().join("\n"),
+        @r###"
```

Comment:
Same here: I think we can remove this sorting by making `jj config list` sort instead.